portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,9 +25,52 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  private ArrayList<String> messages = new ArrayList<>();
+
+  /**
+   * Write to /data the messages ArrayList as a json string
+   */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String json = convertToJson(messages);
+
+    response.setContentType("text/html;");
+    response.getWriter().println(json);
+  }
+
+  /**
+   * Obtain the input from the comment form and add it to the messages ArrayList
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the comment input from the form.
+    String comment = getParameter(request, "text-input", "");
```

Comment:
nit: Add parameter name comment to "" 
https://google.github.io/styleguide/jsguide.html#formatting-param-name-comments

Reply:
Okay.